io-sim/test/Test/IOSim.hs

```diff
@@ -144,8 +146,107 @@ tests =
+  , testGroup "ThreadStatus"
+    [ testProperty "thread status finished (IOSim)"      $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_finished
+    , testProperty "thread status finished (IO)"         $ withMaxSuccess 1 $ ioProperty    prop_thread_status_finished
+    , testProperty "thread status running (IOSim)"       $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_running
+    , testProperty "thread status running (IO)"          $ withMaxSuccess 1 $ ioProperty    prop_thread_status_running
+    , testProperty "thread status blocked (IOSim)"       $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_blocked
+    , testProperty "thread status blocked (IO)"          $ withMaxSuccess 1 $ ioProperty    prop_thread_status_blocked
+    , testProperty "thread status blocked delay (IOSim)" $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_blocked_delay
+    , testProperty "thread status blocked delay (IO)"    $ withMaxSuccess 1 $ ioProperty    prop_thread_status_blocked_delay
+    , testProperty "thread status died (IOSim)"          $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_died
+    , testProperty "thread status died (IO)"             $ withMaxSuccess 1 $ ioProperty    prop_thread_status_died
+    , testProperty "thread status died_own (IOSim)"      $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_died_own
+    , testProperty "thread status died_own (IO)"         $ withMaxSuccess 1 $ ioProperty    prop_thread_status_died_own
+    , testProperty "thread status yield (IOSim)"         $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_yield
+    , testProperty "thread status yield (IO)"            $ withMaxSuccess 1 $ ioProperty    prop_thread_status_yield
+    , testProperty "thread status mask (IOSim)"          $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_mask
+    , testProperty "thread status mask (IO)"             $ withMaxSuccess 1 $ ioProperty    prop_thread_status_mask
+    , testProperty "thread status mask blocked (IOSim)"  $ withMaxSuccess 1 $ runSimOrThrow prop_thread_status_mask_blocked
+    , testProperty "thread status mask blocked (IO)"     $ withMaxSuccess 1 $ ioProperty    prop_thread_status_mask_blocked
+    ]
   ]
 
+--
+-- threadStatus
+--
+
+prop_two_threads_expect :: (MonadFork m, MonadThread m)
+                        => m ()
+                        -> (ThreadId m -> m ())
+                        -> (ThreadStatus -> Property)
+                        -> m Property
+prop_two_threads_expect target main prop = do
+  tid <- forkIO target
+  main tid
+  status <- threadStatus tid
+  return $ prop status
+
+prop_two_threads_expect_ :: (MonadFork m, MonadThread m)
+                         => m ()
+                         -> (ThreadStatus -> Property)
+                         -> m Property
+prop_two_threads_expect_ target prop = prop_two_threads_expect target (const $ yield) prop
+
+prop_thread_status_finished :: (MonadFork m, MonadDelay m, MonadThread m) => m Property
```

Comment:
Could you also try to fit the other type signatures into 80 columns?
```suggestion
prop_thread_status_finished :: (MonadFork m, MonadDelay m, MonadThread m)
                            => m Property
```

Reply:
I need to link our [style guide](https://github.com/input-output-hk/ouroboros-network/blob/master/docs/StyleGuide.md) in this repo.